File: Minimum swaps and K together - GFG/minimum-swaps-and-k-together.py
#User function Template for python3

def minSwap (arr, n, k) : 
    
    count = 0
    
    for i in range(n):
        if arr[i] <= k:
            count += 1
    
    bad = 0
    
    for i in range(count):
        if arr[i] > k:
            bad += 1
    
    ans = bad
    
    j = count
    
    for i in range(n - count ):
        
        # if j == n:
        #     break;
        
        if arr[i] > k: #decreasing count from previous window
            bad -= 1
        
        if arr[j] > k: #increasing count of current window
            bad += 1
        
        j += 1
        
        ans = min(ans , bad)
    
    return ans
        
        
    



#{ 
#  Driver Code Starts
#Initial Template for Python 3

for _ in range(0,int(input())):
    n = int(input())
    arr = list(map(int, input().strip().split()))
    k = int(input())
    ans = minSwap(arr, n, k)
    print(ans)
    





# } Driver Code Ends

File: Minimum swaps and K together - GFG/minimum-swaps-and-k-together.cpp
// { Driver Code Starts
// C++ program to find minimum swaps required
// to club all elements less than or equals
// to k together
#include <bits/stdc++.h>
using namespace std;



 // } Driver Code Ends


class Solution
{
public:
    int minSwap(int arr[], int n, int k) {
        int good = 0;
        for(int i = 0 ; i < n ; i++){
            if(arr[i] <= k) good++;
        }
        
        int bad = 0;
        for(int i = 0 ; i < good;i++){
            if(arr[i] > k) bad++;
        }
        int ans = bad;
        int i = good , j = 0;
        while(i < n){
            if(arr[j] > k) bad--;
            j++;
            if(arr[i] > k) bad++;
            ans = min(ans , bad);
            i+=1;
        }
        return ans;
    }
};


// { Driver Code Starts.

// Driver code
int main() {

	int t,n,k;
	cin>>t;
	while(t--)
    {
        cin>>n;
        int arr[n];
        for(int i=0;i<n;i++)
            cin>>arr[i];
        cin>>k;
        Solution ob;
        cout << ob.minSwap(arr, n, k) << "\n";
    }
	return 0;
}
  // } Driver Code Ends